### ISSUE TITLE
Resource enrollment flows should not default to "auto-discovery"

### ISSUE THREAD

**Issue by r0mant** [gravitational/teleport#44366]
Currently all our resource enrollment flows have "auto-enroll" option turned on by default. Our original reasoning was that "we want users to enroll more resources" but in my opinion it hurts getting started experience. The purpose of Discover flows is to provide easy Day 1 UX where I want to kick the tires and quickly enroll a database that I maybe have or a server, etc. Auto-discovery is not Day 1. I propose we toggle this off by default.

<img width="839" alt="Screenshot 2024-07-17 at 4 01 40 PM" src="https://github.com/user-attachments/assets/10d5eaf7-cdfa-4f1e-8271-1e47855ca3df">

**Comment by roraback** [comment]
What do you think about making this an explicit decision point, instead? I'm thinking something along the lines of: "Do you want to... opt1: enroll a single database, opt2: auto-enroll all databases in a region"

**Comment by r0mant** [comment]
Yeah, I believe we discussed something like this before, including an option to make separate tiles for single resource vs auto-discovery. For now I think just defaulting to auto-discovery "off" is ok.
